engine/class_modules/sc_hunter.cpp

```diff
@@ -3531,10 +3529,21 @@ struct piercing_shot_t: public hunter_ranged_attack_t
   {
     hunter_ranged_attack_t::execute();
 
-    if (result_is_hit(execute_state->result))
+    if (result_is_hit( execute_state -> result))
       trigger_bullseye( p(), execute_state -> action );
 
-    vulnerability_stats.update( p(), execute_state -> action );
+    // execute_state's target may not be the player's target
+    player_t* original_target = nullptr;
+    if ( execute_state -> target != p() -> target )
+    {
+      original_target = execute_state -> target;
+      execute_state -> target = p() -> target;
+    }
+
+    vulnerability_stats.update( p(), execute_state );
+
+    if ( original_target )
+      execute_state -> target = original_target;
```

Comment:
Yeah... looks like I was wrong this whole time.
But, using player target is also wrong, and I believe we should be using action -> target here.